pkg/tesla/module.go

```diff
@@ -6,37 +8,47 @@ import (
 	"fmt"
 
 	"github.com/DIMO-Network/cloudevent"
+	"github.com/DIMO-Network/model-garage/pkg/tesla/api"
+	"github.com/DIMO-Network/model-garage/pkg/tesla/telemetry"
 	"github.com/DIMO-Network/model-garage/pkg/vss"
-	"github.com/tidwall/gjson"
 )
 
 // Module holds dependencies for the Tesla module. At present, there are none.
+// All of the functions on this struct delegate to the appropriate function in
+// the api and telemetry submodules, according to the dataversion field.
 type Module struct{}
 
 // SignalConvert converts a Tesla CloudEvent to DIMO's VSS rows.
 func (m *Module) SignalConvert(_ context.Context, event cloudevent.RawEvent) ([]vss.Signal, error) {
-	return Decode(event)
+	if event.DataVersion == telemetry.DataVersion {
+		return telemetry.SignalConvert(event)
+	}
+	return api.SignalConvert(event)
```

Comment:
This looks clean

Reply:
Could be cleaner if it were more like modules!